src/Handler/StreamHandler.php

```diff
@@ -75,14 +75,20 @@ public function __construct(
         $this->mode = $mode;
         $this->streamOrUrl = $streamOrUrl;
 
+        if (is_a($streamOrUrl, 'XMLParser')) {
+            throw new LogException(__CLASS__ . ': As of php8 XMLparser returns and Object and not a string. Please adapt your stream to the correct string.');
```

Comment:
@boekhorstb1 This exception will also be thrown if you use php<8 right? Is this intended?

Reply:
This exception should not throw in php7, because there XMLParser returns a resource and resources does not have a class ([as of php8 it returns an instance, so then the is_resource() check wont work anymore ](https://php.watch/versions/8.0/XMLParser-resource))

Reply:
Ah very good. My mistake then.